src/pages/parcel/[id].tsx

```diff
@@ -0,0 +1,40 @@
+import getAllParcelData from "@/utils/parcel-fetch";
+import { GetStaticProps, GetStaticPaths, InferGetStaticPropsType } from "next";
+import Parcels from "@/components/Parcels";
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const parcelData = await getAllParcelData();
+
+  // Generate paths based on parcels
+  const paths = parcelData.map((parcel) => ({
+    params: { id: parcel.id.toString() },
+  }));
+
+  // Pre-render only these paths at build time.
+  return { paths, fallback: "blocking" }; // Use 'blocking' for ISR and better user experience
+};
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const id = context.params?.id;
+  const parcelData = await getAllParcelData();
+  const parcel = parcelData.find((p) => p.id.toString() === id);
```

Comment:
It is actually better to extend the parcel fetch to fetch specific id. This way you do not need to do any extra logic here and just present the data as you should.

Also this is not part of the feature and should not be included in this PR